core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/types/UtilTests.kt

```diff
@@ -414,4 +417,44 @@ class UtilTests {
             typeOf<Set<Nothing>?>(),
         ).commonTypeListifyValues() shouldBe typeOf<Collection<out Nothing?>?>()
     }
+
+    @Test
+    fun `common number types`() {
+        // Same type
+        getCommonNumberClass(Int::class, Int::class) shouldBe Int::class
```

Comment:
These tests are great, could you please add a diagramm from your PR here to extract the context about the DAG of type conversion

Reply:
I'll put it somewhere in /documentation so we can include it with KoDEx :)